### PR TITLE
Add support for Python 3.5 - 3.7

### PR DESCRIPTION
**Pull request by sallner**

### CODE REVIEW

**Comment by sallner** [comment]
@icemac Could you please enable the travis check for this repo?

File: setup.py [hunk 1]
```diff
@@ -34,7 +36,7 @@ def read(*rnames):
 
 setup(
     name='z3c.pagelet',
-    version='2.0.1.dev0',
+    version='3.0.dev0',
```
Comment (icemac) [inline setup.py]: Why 3.0? I think 2.1 is enough as it does not break backwards compatibility. Or am I mistaken?

File: CHANGES.rst [hunk 2]
```diff
@@ -2,10 +2,10 @@
 CHANGES
 =======
 
-2.0.1 (unreleased)
-------------------
+2.1 (unreleased)
+----------------
 
-- Nothing changed yet.
+- Add support for Python 3.5 up to 3.7.
```
Comment (icemac) [inline CHANGES.rst]: This could mention, PyPy, too.

File: setup.py (hunk 1 above)
Comment (sallner) [reply setup.py]: We drop 2.6, but 2.1 seems also fine.

File: CHANGES.rst (hunk 2 above)
Comment (sallner) [reply CHANGES.rst]: I forgot to save the changelog before adding. Now it also includes PyPy.

File: src/z3c/pagelet/zcml.py [hunk 3]
```diff
@@ -92,13 +92,6 @@ def pageletDirective(
     # Get the permission; mainly to correctly handle CheckerPublic.
     permission = viewmeta._handle_permission(_context, permission)
 
-    # The class must be specified.
-    if not class_:
-        raise ConfigurationError("Must specify a class.")
-
-    if not zope.interface.interfaces.IInterface.providedBy(provides):
-        raise ConfigurationError("Provides interface provide IInterface.")
-
```
Comment (mgedmin) [inline src/z3c/pagelet/zcml.py]: This doesn't seem to be related to Python version support changes?

Comment (mgedmin) [reply src/z3c/pagelet/zcml.py]: Did you remove dead code after adding those tests and seeing those errors get caught earlier?

Comment (mgedmin) [reply src/z3c/pagelet/zcml.py]: (Ah, yes, I should've looked at the actual commit messages.  I blame GitHub for making commit-by-commit reviews more difficult than all-at-once reviews.)
